### PR TITLE
Fixes for bline

### PR DESCRIPTION
**Pull request by rodolforg**
Fix bugs reported by static analysis tools

### CODE REVIEW

File: synfig-studio/src/gui/states/state_bline.cpp [hunk 1]
```diff
@@ -1852,15 +1856,16 @@ StateBLine_Context::bline_insert_vertex(synfig::ValueNode_Const::Handle value_no
...
-	for(iter=bline_point_list.begin();iter!=bline_point_list.end();++iter)
+	for(list<ValueNode_Const::Handle>::iterator iter=bline_point_list.begin();iter!=bline_point_list.end();++iter)
 		if(*iter==value_node)
 		{
 			bline_point_list.erase(iter);
+			vertex_deleted = true;
 			break;
 		}
-	if(iter==bline_point_list.end())
+	if(vertex_deleted)
```
Comment (ice0) [inline synfig-studio/src/gui/states/state_bline.cpp]: I think it should be !vertex_deleted?

Comment (rodolforg) [reply synfig-studio/src/gui/states/state_bline.cpp]: Sure! Sorry...

Comment (rodolforg) [reply synfig-studio/src/gui/states/state_bline.cpp]: Fixed and rebased to master

**Comment by ice0** [comment]
Merged via https://github.com/synfig/synfig/commit/fc4146f6e76ad9fabbc52c6f583ea74e8a2e0f2c

Thank you!

**Pull request closed without merging**